docs/faqs.md

```diff
@@ -349,3 +338,33 @@ With version ranges the newest compatible package may yield a different package-
 - Build Reproducibility
 
 If consumers try to download and build the recipe at a later time, it may resolve to a different package version that may generate a different binary (that may or may not be compatible). In order to prevent these types of issues, we have decided to only allow exact requirements versions. This is a complicated issue, check [this thread](https://github.com/conan-io/conan-center-index/pull/9140#discussion_r795461547) for more. 
+
+## How to consume a graph of shared libraries?
+
+When the CI builds packages with `shared=True`, it applies the option only to the package being created, but not to
+the requirements. As the default value for the `shared` option is usually `False`, you can expect that the dynamic
+library that has just being generated has linked all its requirements as static libraries.
+
+It is important to remark the default [package id mode](https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#versioning-schema)
+used by Conan (which is the same default used by ConanCenter): `semver_direct_mode`. With this default only the major
+version of the requirements is encoded in the package ID. 
+
+The two previous behaviors together can lead to unexpected results for a user that want to consume a graph of
+dependencies as shared libraries from ConanCenter. They might think that using `*:shared=True` in their profile is
+enough, and indeed Conan will retrieve from ConanCenter all the dynamic libraries for all the graph of dependencies, but
+**all of them will contain the logic of their respective requirements embedded in the dynamic library**, and this
+logic is embedded at the time of building, so it might not match the version of the requirements that was resolved
+by Conan, and for sure, the other dynamic libraries won't be used, only the ones linked directly by the consumer
+project. See a more detailed [example here](https://github.com/conan-io/conan/issues/9712).
+
+In order to consume all those libraries are shared ones, the user needs to build them from sources. This can be
+easily achievable using `*:shared=True` in the _host_ profile and `--build` in the install command. With these inputs,
+Conan will build from sources all the packages and use the shared libraries when linking.
```

Comment:
True, but it can lead to unexpected issues if he wants shared linked to static afterwards since they share the same package id.

Also `--build` is a very strong option, it will build everything from source, including build requirements, even recipes without shared option.

Reply:
I agree with you, but here in this documentation, the best we can do is to try to expose the situation, the risks and the workarounds. If you think something needs to be enforced, just suggest the required changes.

Of course, there is another suggestion, which is to run you own Artifactory and consume your packages only from it. In that Artifactory you can upload only packages built with `*:shared=True` which IMHO would be more expected... But I preferred to focus on the problem than on alternate ConanCenter deployments.